Library/Homebrew/livecheck/strategy.rb

```diff
@@ -180,22 +175,17 @@ def self.page_headers(url, homebrew_curl: false)
         headers = []
 
         [:default, :browser].each do |user_agent|
-          stdout, _, status = curl_with_workarounds(
+          output, _, status = curl_with_workarounds(
             *PAGE_HEADERS_CURL_ARGS, url,
             **DEFAULT_CURL_OPTIONS,
             use_homebrew_curl: homebrew_curl,
             user_agent:        user_agent
           )
+          next unless status.success?
 
-          while stdout.match?(/\AHTTP.*\r$/)
-            h, stdout = stdout.split("\r\n\r\n", 2)
-
-            headers << h.split("\r\n").drop(1)
-                        .to_h { |header| header.split(/:\s*/, 2) }
-                        .transform_keys(&:downcase)
-          end
-
-          return headers if status.success?
+          parsed_output = parse_curl_output(output)
+          parsed_output[:responses].each { |response| headers << response[:headers] }
+          return headers if headers.present?
```

Comment:
```suggestion
          break if headers.present?
```

as you're returning it afterwards anyway